http/codegen/openapi/json_schema.go

```diff
@@ -434,34 +459,14 @@ func (s *Schema) Dup() *Schema {
 // buildAttributeSchema initializes the given JSON schema that corresponds to
 // the given attribute.
 func buildAttributeSchema(api *design.APIExpr, s *Schema, at *design.AttributeExpr) *Schema {
-	s.Merge(TypeSchema(api, at.Type))
+	s.Merge(TypeSchema(api, at.Type, at.Validation))
 	if s.Ref != "" {
 		// Ref is exclusive with other fields
 		return s
 	}
 	s.DefaultValue = toStringMap(at.DefaultValue)
 	s.Description = at.Description
 	s.Example = at.Example(api.Random())
-	val := at.Validation
```

Comment:
See comment above. This seems to be the correct place to go through the validations on the AttributeExpr.

Reply:
@nitinmohan87 Look at this:

```
http/codegen/openapi/openapi_v2_builder.go
427:            schema = TypeSchema(root.Design.API, r.Body.Type, r.Body.Validation)
491:                    schema := TypeSchema(root.Design.API, design.ErrorResult, nil)
569:                            Schema:      TypeSchema(root.Design.API, endpoint.Body.Type, endpoint.Body.Validation),

http/codegen/openapi/json_schema.go
171:                    requestSchema = TypeSchema(api, a.MethodExpr.Payload.Type, a.MethodExpr.Payload.Validation)
196:                                    targetSchema = TypeSchema(api, mt, val)
200:                                    targetSchema.AnyOf = []*Schema{firstSchema, TypeSchema(api, mt, val)}
202:                                    targetSchema.AnyOf = append(targetSchema.AnyOf, TypeSchema(api, mt, val))
462:    s.Merge(TypeSchema(api, at.Type, at.Validation))

```

In all that places I need to copy the validations. If someone adds a TypeSchema call, he will have to add the validations too. 

If you know a better option, I'm ready to change it.

Reply:
I think we can probably export this `buildAttributeSchema` to `AttributeSchema` and call this wherever `TypeSchema` is called? The validation logic shall remain in the `AttributeSchema` since validations are tied to an AttributeExpr.

Probably these lines can still call `TypeSchema`
```
491:                    schema := TypeSchema(root.Design.API, design.ErrorResult, nil)
```
Makes sense?

Reply:
OK, I'll refactor the ```buildAttributeSchema```.